app/graphql/types/rule.rb

```diff
@@ -24,21 +24,27 @@ class Rule < Types::BaseObject
     end
 
     def compliant(args = {})
-      system_id(args) &&
-        profile_id(args) &&
-        %w[pass notapplicable notselected].include?(
-          context[:rule_results][object.id][profile_id(args)]
-        )
+      ::Rails.cache.fetch(
+        "#{system_id(args)}/#{profile_id(args)}/#{object.id}/compliant"
```

Comment:
Ok, this one does make sense, since multiple systems can be tested against a single rule+profile combination.